### PR TITLE
MM-9958 UI updates to personal access token creation

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
This includes two UI updates for personal access tokens:
* Add a spinner to the create button
* Add a modal when a user tries to navigate away from the modal after creating a token, used to make sure the user has copied the token

Reviewing developers, I also factored out the access token section of the security settings into it's own component.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-9958

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

**Comment by jwilander** [comment]
@jasonblais some notes:
* Please look at the text for the modal, feel free to make changes there without asking
* You can throttle your connection to get a slow response to see the loading state of the modal
* I didn't use a spinner for the modal when the token wasn't loaded yet because I don't think we have an existing one that really fit there, but let me know if you want me to change it to something
* On the modal I hid the cancel button but I'm not sure if the experience is quite right for when the token is still loading, since the text doesn't quite make sense. Let me know what you think
* The user can still close the entire browser window and the modal won't appear, I can probably have it appear in that case as well but I think that might be overkill

**Comment by jasonblais** [comment]
@jwilander Sorry for the delay - have tested and have two proposed UI changes. I'll run it by the UX team first since I'm only 3/5 on the UI changes. We'll post back here once I've heard their feedback.

Otherwise looks good to me. Also agree that we don't need to cover the case when the user closed their browser window - in that case they can just create a new access token.

**Comment by jasonblais** [comment]
@jwilander Just two proposals:

1. Remove yellow background and warning icon to match with other similar modals like deleting messages or channels.

2. Replace `Loading…` on second screen with `Access Token: Loading…` where `Access Token` is bolded.

These changes were approved. Let me know if you have questions about them.

**Comment by jwilander** [comment]
Sounds good, I'll make those changes

**Comment by jwilander** [comment]
@jasonblais changes made

File: components/user_settings/security/user_access_token_section/user_access_token_section.jsx [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+                return;
+            }
+
+            let activeLink;
+            let activeStatus;
+
+            if (token.is_active) {
+                activeLink = (
+                    <a
+                        name={token.id + '_deactivate'}
```
Comment (lieut-data) [inline components/user_settings/security/user_access_token_section/user_access_token_section.jsx]: I know this is from the old code, but is this `name` used anywhere?

File: components/user_settings/security/user_access_token_section/user_access_token_section.jsx [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+                        onClick={(e) => {
+                            e.preventDefault();
+                            this.deactivateToken(token.id);
+                        }}
+                    >
+                        <FormattedMessage
+                            id='user.settings.tokens.deactivate'
+                            defaultMessage='Deactivate'
+                        />
+                    </a>);
```
Comment (lieut-data) [inline components/user_settings/security/user_access_token_section/user_access_token_section.jsx]: nit: This is also probably from the old code, but the parentheses ends on a different line than line 302 would suggest.

**Pull request merged**